python/lancedb/fts.py

```diff
@@ -56,7 +56,12 @@ def create_index(index_path: str, text_fields: List[str]) -> tantivy.Index:
     return index
 
 
-def populate_index(index: tantivy.Index, table: LanceTable, fields: List[str]) -> int:
+def populate_index(
+    index: tantivy.Index,
+    table: LanceTable,
+    fields: List[str],
+    writer_heap_size: Optional[int] = 1024 * 1024 * 1024,
```

Comment:
`Optional` implies you can pass `None`, which I think would error in tantivy:

https://github.com/quickwit-oss/tantivy-py/blob/0cf1661fb371c8a9c12050ac378a187223ef50af/src/index.rs#L265-L270

```suggestion
    writer_heap_size: int = 1024 * 1024 * 1024,
```